Seeds : Add primitiveVariables plug

danieldresser-ie opened the pull request:
Simple commit to expose the feature from https://github.com/ImageEngine/cortex/pull/1353

danieldresser-ie commented:
Added commit to rename Seeds -> Scatter.

johnhaddon commented on startup/GafferScene/seedsCompatibility.py:
```diff
@@ -0,0 +1,41 @@
+import GafferScene
+
+GafferScene.Seeds = GafferScene.Scatter
+Gaffer.Metadata.registerValue( GafferScene.Scatter, "name", "userDefault", "scatter" )
```
Maybe `points` is a better default, because scene locations should be nouns?

johnhaddon commented on python/GafferSceneUI/SeedsUI.py:
```diff
@@ -99,6 +99,16 @@
+
+			"description",
+			"""
+			Primitive variables to output on the generated points. Supports a Gaffer match pattern,
```
Should we make a point of saying that they are sampled from the surface?

johnhaddon reviewed (commented):
> Added commit to rename Seeds -> Scatter.

This needs noting as a breaking change, and will need to be merged to `main` and not `1.2_maintenance`. Can we split it into a separate PR please?

danieldresser-ie commented:
Now that we have decided to put the Cortex changes in 10.5, this whole PR needs to target main, right? So we don't need to split it up?

I've addressed feedback, plus exposed refPosition now that we've decided it is actually useful, and snuck in a change to config/ie/options so that the Sconstruct doesn't fail out here.

danieldresser-ie commented:
Renamed refPositions to referencePositions and added uv plug

johnhaddon commented:
I've merged #5301, so if you rebase this on top of the latest `main`, the CI should go through now.

danieldresser-ie commented:
Rebased.

johnhaddon commented:
I've merged, and separately I've updated Changes.md.
